### PR TITLE
Closes #1962 #1969 admiral 0.11.1 hotfix for get started link, derive_param_tte, derive_vars_joined

### LINKED ISSUE #1962: Bug: `derive_param_tte()` fails for `dplyr 1.1.0`
### What happened?

If `dlpyr 1.1.0` is used, some tests for `derive_param_tte()` fail. They pass for `dplyr 1.0.5` (minimum version required by `admiraldev`) and also for `dplyr 1.1.1`.

In `derive_param_tte()`
```
        full_join(
          mutate(by_groups, temp_dummy = 1),
          mutate(source_datasets[[i]], temp_dummy = 1),
          by = "temp_dummy",
          relationship = "many-to-many"
        ) %>%
        select(-temp_dummy)
```
should be replaced by
```
        crossing(
          by_groups,
          source_datasets[[i]]
        )
```
This is simpler and avoids the `dplyr` issue.

### Session Information

_No response_

### Reproducible Example

_No response_

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_joined.R [hunk 1]
```diff
@@ -352,6 +352,10 @@ derive_vars_joined <- function(dataset,
 
   if (is.null(new_vars)) {
     new_vars <- chr2vars(colnames(dataset_add))
+    if (any(colnames(dataset_add) %in% colnames(dataset))) {
+      warn_if_vars_exist(dataset, new_vars)
+      warn("Be careful to set appropriate expressions for `by_vars`, `join_vars` and `new_vars`.")
```
Comment (zdz2101) [inline R/derive_joined.R:357]: @bundfussr Do you think this is an appropriate way to address the NULL `new_vars` piece? What're the example use-cases of using this function without `new_vars`

File: R/derive_joined.R [hunk 2]
```diff
@@ -371,7 +375,7 @@ derive_vars_joined <- function(dataset,
     filter_if(filter_add) %>%
     select(
       !!!by_vars,
-      !!!chr2vars(names(order)),
+      !!!extract_vars(order),
```
Comment (zdz2101) [inline R/derive_joined.R:378]: I think this fixed the second part of the problem though @bundfussr or at least I'm no longer seeing the duplicates warning

File: R/derive_joined.R (hunk 1 above)
Comment (zdz2101) [reply R/derive_joined.R:357]: A really hacky solution would be to insert like at the end
if (is.null(new_vars)){
  return(
    data %>%
     select(-ends_with(".join")) 
  ) 
}

File: R/derive_joined.R (hunk 2 above)
Comment (zdz2101) [reply R/derive_joined.R:378]: Nvm it broke a couple tests

File: R/call_derivation.R
Comment (bms63) [inline R/call_derivation.R:1]: IS this needed for the bug fix?

File: R/derive_basetype_records.R
Comment (bms63) [inline R/derive_basetype_records.R:1]: IS this needed for the bug fix?

File: R/derive_param_tte.R [hunk 3]
```diff
@@ -738,14 +738,7 @@ extend_source_datasets <- function(source_datasets,
   by_groups <- unique(bind_rows(by_groups))
   for (i in seq_along(source_datasets)) {
     if (extend[[i]]) {
-      source_datasets[[i]] <-
-        full_join(
-          mutate(by_groups, temp_dummy = 1),
-          mutate(source_datasets[[i]], temp_dummy = 1),
-          by = "temp_dummy",
-          relationship = "many-to-many"
```
Comment (bms63) [inline R/derive_param_tte.R:746]: does this impact the `admiral_environment` stuff??

File: R/derive_basetype_records.R
Comment (zdz2101) [reply R/derive_basetype_records.R:1]: I was getting ci/cd errors related to the assertion stuff which is why I went ahead and just copied over the same changes

File: R/derive_param_tte.R (hunk 3 above)
Comment (zdz2101) [reply R/derive_param_tte.R:746]: From my understanding it shoudn't, only objects in there right now are "many-to-one" , "one-to-many" , the subject keys stuff, and duplicates

File: R/derive_joined.R [hunk 4]
```diff
@@ -371,7 +375,7 @@ derive_vars_joined <- function(dataset,
     filter_if(filter_add) %>%
     select(
       !!!by_vars,
-      !!!chr2vars(names(order)),
+      !!!replace_values_by_names(order),
```
Comment (zdz2101) [inline R/derive_joined.R:378]: This fixed the duplicates messaging in the reprex in the issue, but to make it work needed to assign the functions calls column names inside the order argument

File: R/derive_param_tte.R (hunk 3 above)
Comment (bundfussr) [reply R/derive_param_tte.R:746]: > does this impact the `admiral_environment` stuff??

No, this is something completely different.

File: R/derive_joined.R (hunk 4 above)
Comment (bundfussr) [reply R/derive_joined.R:378]: We should not force the users to name the expressions specified for `order`.

File: vignettes/generic.Rmd [hunk 5]
```diff
@@ -313,7 +313,7 @@ ae_02 <- ae_01 %>%
     dataset_add = ae_01,
     filter_add = AESTDY > 0,
     by_vars = exprs(USUBJID),
-    order = exprs(as.integer(factor(AESEV, levels = c("SEVERE", "MODERATE", "MILD")))),
+    order = exprs(AESEV = as.integer(factor(AESEV, levels = c("SEVERE", "MODERATE", "MILD")))),
```
Comment (bundfussr) [inline vignettes/generic.Rmd:316]: This changes the output dataset. The bug fix for `derive_vars_joined()` should not require to change the calls to `derive_vars_joined()`.

**Review by bundfussr — COMMENTED** [review]
I would remove the fix of `derive_vars_joined()` from the hotfix. It seems that it requires more time. I am a bit concerned that the fix introduce another bug or a break of backward compatibility.

Some changes, e.g., in `derive_basetype_records()` look like related to deprecation of assertions in the next admiraldev version. Is this intended?

**Review by bundfussr — APPROVED** [review]

**Pull request merged**